geometry/create-and-edit-geometries/src/main/java/com/esri/samples/create_and_edit_geometries/CreateAndEditGeometriesController.java

```diff
@@ -0,0 +1,451 @@
+      pointGraphic.setSymbol(pointSymbol);
+      multipointGraphic.setSymbol(multipointSymbol);
+      polyline1Graphic.setSymbol(lineSymbol);
+      polyline2Graphic.setSymbol(lineSymbol);
+      polygonGraphic.setSymbol(fillSymbol);
+
+      // add example geometry graphics to the graphics overlay
+      graphicsOverlay.getGraphics().addAll(List.of(pointGraphic, multipointGraphic, polyline1Graphic,
+        polyline2Graphic, polygonGraphic));
+    } catch (Exception e) {
+      new Alert(Alert.AlertType.ERROR, "Unable to load example geometries from file").show();
+    }
+
+  }
+
+  /**
+   * Starts the geometry editor with the point geometry type.
+   */
+  @FXML
+  private void handlePointButtonClicked() {
+    if (!geometryEditor.isStarted()) {
+      graphicsOverlay.clearSelection();
+      geometryEditor.setTool(vertexTool);
+      geometryEditor.start(GeometryType.POINT);
+    }
+  }
+
+  /**
+   * Starts the geometry editor with the multipoint geometry type.
+   */
+  @FXML
+  private void handleMultipointButtonClicked() {
+    if (!geometryEditor.isStarted()) {
+      graphicsOverlay.clearSelection();
+      geometryEditor.setTool(vertexTool);
+      geometryEditor.start(GeometryType.MULTIPOINT);
+    }
+  }
+
+  /**
+   * Starts the geometry editor with the polyline geometry type.
+   */
+  @FXML
+  private void handlePolylineButtonClicked() {
+    if (!geometryEditor.isStarted()) {
+      graphicsOverlay.clearSelection();
+      geometryEditor.start(GeometryType.POLYLINE);
+    }
+  }
+
+  /**
+   * Starts the geometry editor with the polygon geometry type.
+   */
+  @FXML
+  private void handlePolygonButtonClicked() {
+    if (!geometryEditor.isStarted()) {
+      graphicsOverlay.clearSelection();
+      geometryEditor.start(GeometryType.POLYGON);
+    }
+  }
+
+  /**
+   * Undo the last change made to the geometry while editing is active.
+   */
+  @FXML
+  private void handleUndoButtonClicked() {
+    geometryEditor.undo();
+  }
+
+  /**
+   * Redo the last change made to the geometry while editing is active.
+   */
+  @FXML
+  private void handleRedoButtonClicked() {
+    geometryEditor.redo();
+  }
+
+  /**
+   * Deletes the currently selected element of the geometry.
+   */
+  @FXML
+  private void handleDeleteSelectedElementButtonClicked() {
+    geometryEditor.deleteSelectedElement();
+  }
+
+  /**
+   * Saves the geometry in the editor to the graphics overlay and sets its symbol to the type relevant for the geometry.
+   */
+  @FXML
+  private void handleStopAndSaveButtonClicked() {
+
+    // handle saving depending on if the geometry being edited is new, or already exists and needs updating
+    if (!geometryEditor.getGeometry().isEmpty()) {
+      if (selectedGraphic != null) {
+        updateSelectedGraphic();
+      } else {
+        createNewGraphic();
+      }
+    } else {
+      new Alert(Alert.AlertType.ERROR, "Geometry is empty. \nUnable to save geometry.").show();
+    }
```

Comment:
```suggestion
      if (selectedGraphic != null) {
        updateSelectedGraphic();
      } else {
        createNewGraphic();
      }
```
Indentation will need adjusting here too.

This check is actually not good, as we need to be able to save when we delete a selected element. Currently this is a bug preventing us from doing so. It's following the implementation details, so I'll need to query it. For now though we should remove it.